blocks/library/reusable-block/index.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { pickBy, noop } from 'lodash';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Placeholder, Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockType, registerBlockType } from '../../api';
+import ReusableBlockEditPanel from './edit-panel';
+
+class ReusableBlockEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.startEditing = this.startEditing.bind( this );
+		this.stopEditing = this.stopEditing.bind( this );
+		this.setAttributes = this.setAttributes.bind( this );
+		this.setName = this.setName.bind( this );
+		this.updateReusableBlock = this.updateReusableBlock.bind( this );
+
+		this.state = {
+			isEditing: false,
+			name: null,
+			attributes: null,
+		};
+	}
+
+	componentDidMount() {
+		if ( ! this.props.reusableBlock ) {
+			this.props.fetchReusableBlock();
```

Comment:
It *would* be nice to simplify this all with some abstraction. I'm not sure that `withAPIData` is the right abstraction, though.

The major problem with using `withAPIData` is that our 'Detach from Reusable Block' functionality needs to have the attributes of the reusable block that it is detaching. Currently, it gets this by [accessing the store](https://github.com/WordPress/gutenberg/pull/2659/files#diff-c0e459b150c84795a6a30e7104d1027fR349). If we moved the fetched reusable block to inside the `edit` component's local state, then we would not be able to access this from outside of the component.

Also: right now, if you have two instances of the same reusable block in the post, editing one will immediately update the other. We get this for free because the UI is derived from what's in the store. It's a small thing, but kind of nice.